service/validator.go

```diff
@@ -0,0 +1,49 @@
+package service
+
+import (
+	"fmt"
+	"github.com/streamingfast/substreams/manifest"
+	pbsubstreams "github.com/streamingfast/substreams/pb/sf/substreams/v1"
+)
+
+type RequestValidator struct {
+	blockType string
+}
+
+func NewRequestValidator(blockType string) *RequestValidator {
+	return &RequestValidator{
+		blockType: blockType,
+	}
+}
+
+func (r *RequestValidator) Validate(request *pbsubstreams.Request) (*manifest.ModuleGraph, error) {
```

Comment:
was the goal of this object not to have to store `blockType` on the `Service` object anymore?
Couldn't the `Validate()` method simply be passed the `blockType` from `s.blockType` ?

If this can be simply a function, that's easier than an object, especially if it's just to enclose a variable that is available at the moment of calling `Validate()`.